arrows/serialize/json/load_save.cxx

```diff
@@ -646,15 +655,54 @@ void load( cereal::JSONInputArchive& archive,
            CEREAL_NVP( confidence ) );
   load( archive, start_frame );
   load( archive, end_frame );
-  load( archive, *class_map );
-  load( archive, *participants );
   activity.set_id( id );
   activity.set_label( label );
   activity.set_confidence( confidence );
-  activity.set_activity_type( class_map );
   activity.set_start( start_frame );
   activity.set_end( end_frame );
-  activity.set_participants( participants );
+
+  // Optional parameters aren't supported
+  // for JSON. activity_type and participants may or may not exist,
+  // so we check if an exception is thrown when we look for them.
+  // If an exception is thrown, we don't set those fields.
+  bool has_act_type = true;
+  bool has_participants = true;
+
+  try
+  {
+    load( archive, *act_type );
+  }
+  catch( cereal::Exception& )
+  {
+    has_act_type = false;
+  }
+
+  try
+  {
+    load( archive, *participants );
+  }
+  catch( cereal::Exception& )
+  {
+    has_participants = false;
+  }
+
+  if ( has_act_type )
+  {
+    activity.set_activity_type( act_type );
+  }
+  else
+  {
+    activity.set_activity_type( nullptr );
```

Comment:
Can we set activity type to nullptr [here](https://github.com/Kitware/kwiver/pull/1024/files#diff-09735ca8ba7ffc48e8004498c9f7d14bR677)?

Reply:
Yes, great idea.

Reply:
Fixed in 560077cce47e58452637078afd1b6d92ed82c0ef